packages/atomic/cypress/utils/analyticsUtils.ts

```diff
@@ -0,0 +1,79 @@
+import {
+  SearchEventRequest,
+  ClickEventRequest,
+  CustomEventRequest,
+} from 'coveo.analytics/src/events';
+import {findLast} from './arrayUtils';
+
+export type AnyEventRequest =
+  | SearchEventRequest
+  | ClickEventRequest
+  | CustomEventRequest;
+
+function isSearchEventRequest(
+  request: AnyEventRequest
+): request is SearchEventRequest {
+  return 'actionCause' in request;
+}
+
+function isClickEventRequest(
+  request: AnyEventRequest
+): request is ClickEventRequest {
+  return 'documentUri' in request;
+}
+
+function isCustomEventRequest(
+  request: AnyEventRequest
+): request is CustomEventRequest {
+  return 'eventType' in request;
+}
+
+const analyticsEventsKey = '_analyticsEvents';
+
+interface WindowWithAnalytics extends Window {
+  [analyticsEventsKey]?: AnyEventRequest[];
+}
+
+export class AnalyticsTracker {
+  private static get window() {
+    return window as WindowWithAnalytics;
+  }
+
+  private static get analytics() {
+    return (this.window[analyticsEventsKey] =
+      this.window[analyticsEventsKey] ?? []);
+  }
+
+  private static set analytics(analytics: AnyEventRequest[]) {
+    this.window[analyticsEventsKey] = analytics;
+  }
+
+  static push(request: AnyEventRequest) {
+    this.analytics.push(request);
+  }
+
+  static reset() {
+    this.analytics = [];
+  }
```

Comment:
For now this is only used in `TestFixture`, but I think it could also be useful in some other cases (e.g., clicking the same button multiple times and checking that the last also sent analytics).